tensorboard/plugins/mesh/plugin_data.proto

```diff
@@ -12,7 +12,7 @@ message MeshPluginData {
     COLOR = 3;
   }
 
-  // Version `0` is the only supported version.
+  // Version `1` is the only supported version.
```

Comment:
Not really, the JS won't make any requests to the server if `components=0`

Reply:
Okay: then `parse_plugin_metadata` should be updated to translate from
the old format to the new format, if at all possible. Is it the case
that a version-0 summary should be interpreted as if its `components`
field were set to the union of all the content types? If so, then
consider something like:

```python
def get_components_bitmask(content_types):
  """<snip>"""
  result = 0
  for content_type in content_types:
    if content_type == plugin_data_pb2.MeshPluginData.ContentType.UNDEFINED:
      raise ValueError('Cannot include UNDEFINED content type in mask')
    result |= (1 << content_type)
  return result


_DEFAULT_COMPONENTS_MASK = get_components_bitmask([
    plugin_data_pb2.MeshPluginData.ContentType.VERTEX,
    plugin_data_pb2.MeshPluginData.ContentType.FACE,
    plugin_data_pb2.MeshPluginData.ContentType.COLOR,
])


def parse_plugin_metadata(content):
  """<snip>"""
  if not isinstance(content, bytes):
    raise TypeError('Content type must be bytes.')
  result = plugin_data_pb2.MeshPluginData.FromString(content)
  if not 0 <= result.version <= get_current_version():
    raise ValueError('Unknown metadata version: %s. The latest version known '
                     'to this build of TensorBoard is %s; perhaps a newer '
                     'build is available?' %
                     (result.version, get_current_version()))
  if result.version < 1:
    result.components = _DEFAULT_COMPONENTS_MASK
  result.version = get_current_version()
  return result
```

(This is the reason that the `parse_plugin_metadata` functions and the
`version` field exist in the first place: to provide a translation layer
for old data rather than just choking on it. We just announced the mesh
plugin as part of TensorBoard 1.14; we can’t immediately turn around and
drop data compatibility for the summaries that people have now written.)

Another way to solve this problem would be to change the `components`
field to a `missing_components` field representing a bitmask of the
elements that are _not_ included in the summary, such that the zero
value is useful. This would also have the benefit of saving 5 bytes per
summary. Your summaries would be backward-compatible for free, and you
wouldn’t need to bump the version. The two downsides that I see are that
(a) it’s slightly surprising, and (b) it makes it harder to add new
content types down the line.

Reply:
I think interpreting 0 as "include all" makes more sense, since it is not possible to have 0 value after current PR. `UNDEFINED` is not being currently used.